### DISCUSSION TITLE
learner/user/v1/signup api working on local but giving whitelist error on prod

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#534] [Bugs]
Hello Team,
Hope you are doing well!..

I am doing some custom flow to user signup in my project
For that I used **/learner/user/v1/signup** API. Its working fine for locahost but giving ** "Forbidden: API WHITELIST Access is denied"** error on production.
Any idea why it happens and whats the solution?
Can anyone help me here?

Working curl:

`curl 'http://localhost:3000/learner/user/v1/signup' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: somecokie' \
  -H 'Origin: http://localhost:3000' \
  -H 'Referer: http://localhost:3000/signup?client_id=portal&state=a1eab6e8-7979-492e-865b-38276ab4fefe&redirect_uri=http:%2F%2Flocalhost:3000%2Fresources%3Registration%26auth_callback%3D1&scope=openid&response_type=code&version=4&error_callback=https:%2F%2Fnulp.niua.org%2Fauth%2Frealms%2Fsunbird%2Fprotocol%2Fopenid-connect%2Fauth' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: local.sunbird.portal' \
  -H 'X-App-Version: 4.4.1' \
  -H 'X-Channel-Id: 0134866036584857600' \
  -H 'X-Device-ID: devId' \
  -H 'X-Org-code: 0134866036584857600' \
  -H 'X-Request-ID: reqId' \
  -H 'X-Session-ID: sessionId' \
  -H 'X-Source: web' \
  -H 'X-msgid: 2b37bf11-2801-cb86-4bd8-5246e43473c0' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-06-13T15:34:00+05:30' \
  --data-raw '{"params":{"source":"portal","signupType":"self"},"request":{"firstName":"testUser","password":"Password@123","dob":"2020","email":"testUser@yopmail.com","emailVerified":true,"channel":"nulp-sbm","framework":{"state":["Karnataka"],"urbanLocalBody":["test"],"designation":["test"],"id":["framid"]}}}' \
  --compressed`

Non Working curl:
`curl 'https://prodUrl/learner/user/v1/signup' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: somecokie' \
  -H 'Origin: https://prodUrl' \
  -H 'Referer: https://prodUrl/signup?client_id=portal&state=ed8cee47-484a-4915-b414-6eb5d5748098&redirect_uri=https:%2F%2FprodUrl%2Fresources%3Registration%26auth_callback%3D1&scope=openid&response_type=code&version=4&error_callback=https:%2F%2Fnulp.niua.org%2Fauth%2Frealms%2Fsunbird%2Fprotocol%2Fopenid-connect%2Fauth' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: prodUrl.portal' \
  -H 'X-App-Version: 4.4.1' \
  -H 'X-Channel-Id: 0134866036584857600' \
  -H 'X-Device-ID: devId' \
  -H 'X-Org-code: 0134866036584857600' \
  -H 'X-Request-ID: d3fe6834-fecb-d54a-444e-d71fc3246b3c' \
  -H 'X-Session-ID: d5657443-da8c-f41c-845c-80aa04df6aae' \
  -H 'X-Source: web' \
  -H 'X-msgid: d3fe6834-fecb-d54a-444e-d71fc3246b3c' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-06-13T10:34:03+05:30' \
  --data-raw '{"params":{"source":"portal","signupType":"self"},"request":{"firstName":"Komal Mane","password":"Password@123","dob":"1993","email":"komal_sbm1@yopmail.com","emailVerified":true,"channel":"nulp-sbm","framework":{"state":["test"],"urbanLocalBody":["Test"],"designation":["ttest"],"id":["framidtes"]},"reqData":"{\"iv\":\"a03e290aa09c4930e9d04c589b07457c\",\"encryptedData\":\"1213f2165a75e9e1a7ed5780ab7ef0435b73a05424c8b9c09a5167fa1924b4d2a0dc52c2398da82eb450bdb7236c14af8434289a8445fcde9ffea18fa2c6e7b9\"}"}}' \
  --compressed`

cc: @coolbung, @maheshkumargangula, @amitpriyadarshi, @keshavprasadms, @pallakartheekreddy,

**Comment by paritshivani** [comment]
Can anyone help me here?

**Reply by Krishnaj20** [reply]
@rajeevsathish - can you look into this thread and tag the right people to respond to this discussion.

**Comment by amitpriyadarshi** [comment]
@paritshivani can you plz confirm once whether the api on onboard is done or not?
@sharathprasad4tarento As per my understanding it is onboarding specific issue. Can you suggest possible reason?

**Reply by amitpriyadarshi** [reply]
@gandham-santhosh @tsprasath it seems this api has some onboarding issue. This api onboarding has not happened  or not happened properly. Can you please comment on this.
cc: @Krishnaj20 @vinukumar-vs

**Reply by gandham-santhosh** [reply]
@paritshivani can you please let us know if this issue is resolved?

**Reply by tsprasath** [reply]
@paritshivani Please ssh to  apimanager pod and check the api is available  `curl  -s localhost:8001/apis?size=8192 | python -m json.tool | grep learner/user/v1/signup`

**Reply by chitranshu-keshav** [reply]
@paritshivani can you please get back incase your issue is not resolved. If its resolved, please mark the thread as answered
cc @ashokreddy1208
